File: samples/v2/csharp/ContactsTest.cs
namespace Keap.Core.V2.Tests;

using Keap.Core.V2.Api;
using Keap.Core.V2.Client;

[TestClass]
public class ContactsTest
{
    [TestMethod]
    public void ListContactsSucceeds()
    {
        var apiKey = Environment.GetEnvironmentVariable("KEAP_REST_API_SERVICE_ACCESS_TOKEN") ?? throw new InvalidOperationException("KEAP_REST_API_SERVICE_ACCESS_TOKEN is not set");

        var config = new Configuration();
        config.DefaultHeaders["Authorization"] = $"Bearer {apiKey}";

        var contactApi = new ContactApi(config);
        var response = contactApi.ListContactsUsingGET1();

        Console.WriteLine($"Found {response.Contacts.Count()} contacts");
        foreach (var contact in response.Contacts)
        {
            Assert.IsNotNull(contact.Id, $"Contact has no ID");
        }
    }
}

File: samples/v2/python/contacts_test.py
import os
import unittest
from keap_core_v2_client import ApiClient, ContactApi, Configuration

class ContactsTest(unittest.TestCase):

    def test_list_contacts_succeeds(self):
        api_key = os.getenv('KEAP_REST_API_SERVICE_ACCESS_TOKEN')
        if not api_key:
            raise EnvironmentError('KEAP_REST_API_SERVICE_ACCESS_TOKEN is not set')

        config = Configuration()
        config.api_key['api_key'] = api_key

        with ApiClient(configuration=config) as api_client:
            api_client.default_headers['Authorization'] = f"Bearer {api_key}"
            contact_api = ContactApi(api_client)
            response = contact_api.list_contacts_using_get1()

            print(f"Found {len(response.contacts)} contacts")
            for contact in response.contacts:
                self.assertIsNotNone(contact.id)

if __name__ == '__main__':
    unittest.main()
